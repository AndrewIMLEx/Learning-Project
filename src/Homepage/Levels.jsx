import React, { Component} from "react"

class Levels extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            levelSelected: "LEVEL",             
            input: "",
            tooHightooLow: "",
            standardNum: Math.floor(Math.random() *10) + 1 + "",
            difficultNum: Math.floor(Math.random() *100) + 1 + "",
        };
        this.standard = this.standard.bind(this);
        this.difficult = this.difficult.bind(this);
        this.guessNumber = this.guessNumber.bind(this);
        this.clickedbuttonGuess = this.clickedbuttonGuess.bind(this);
        this.Initial = this.Initial.bind(this)
    }
    Initial(){
        this.setState({
            levelSelected: "LEVEL",             
            input: "",
            tooHightooLow: "",
            guessNumber: "",
            standardNum : Math.floor(Math.random() *10) + 1 + "",
            difficultNum: Math.floor(Math.random() * 100 + 1) + "",         
    })
    }
    standard(e){
        e.preventDefault();
        this.setState({levelSelected:'STANDARD'});
    }
    difficult(e){
        e.preventDefault();
        this.setState({levelSelected:'DIFFICULT'});
    }
    componentDidMount(){
        if(this.standard){
            this.setState({standardNum : Math.floor(Math.random() *10) + 1 + ""})
        }
        else if(this.difficult){
            this.setState({difficultNum: Math.floor(Math.random() *100) + 1 + ""})
        }
    }
    componentDidUpdate(){
      
    }
    guessNumber(e){
        e.preventDefault();
        this.setState({ input:e.target.value})
    }
clickedbuttonGuess(e){
    e.preventDefault(); 
    let startNewStandardGame = this.state.standardNum;
    let startNewDifficultGame = this.state.difficultNum;
    let guessedValue = this.guessNumber;  

    console.log(this.state.levelSelected);
    console.log(this.state.input);

        if(guessedValue == startNewStandardGame){
            this.setState({tooHightooLow:'Good Job!Correct'});
        }else if(guessedValue < startNewStandardGame){
            this.setState({tooHightooLow:'Wrong!Too Low'});
        }else if(guessedValue  > startNewStandardGame){
            this.setState({tooHightooLow:'Wrong!Too high'});
        }
        if(guessedValue  == startNewDifficultGame){
            this.setState({tooHightooLow:'Good Job!Correct'});
        }else if(guessedValue  > startNewDifficultGame){
            this.setState({tooHightooLow:'Wrong!too High'});
        }else if(guessedValue  < startNewDifficultGame){
            this.setState({tooHightooLow:'Wrong!Too Low'});
        }
        else {
            this.setState({tooHightooLow:'PLEASE ENTER A NUMBER'});
        }
    }
render(){
    return(
        <>
        <div className="levelselect">
        <h5 style={{color: "white",fontSize:"17px",marginLeft:'5em'}}>Select Difficulty Level</h5>
        <button id="standard" value={this.state.standard}  onClick ={this.standard}>Standard</button>
        <button id="difficult" value={this.state.difficult} onClick ={this.difficult}>Difficult</button>
        </div>
        <div className="levelCard">
        <h3 style={{color: "black",fontSize: '20px',marginLeft: '6em'}}>Selected Level</h3>
        &nbsp;
        <p style={{color: "red",fontSize: 20,marginLeft:'6.9em'}}>{ this.state.levelSelected }</p>
        &nbsp;
        <form id = "LevelForm">
            <input style={{float:'center'}} value={this.state.input}  id = "Levelsinput" placeholder="Enter Number Here" type="number" onChange = {this.guessNumber} ref={this.inputFocus}/>
        </form>
        &nbsp;&nbsp;
        <p style={{color: "red",marginLeft:'6.5em'}}>{this.state.tooHightooLow}</p>
        &nbsp;&nbsp;
        <button onClick={this.clickedbuttonGuess}  id="guess" type = "submit">Guess</button>
        &nbsp;
        <div className="playButtons">
        <button id="play"  onClick={this.Initial} type = "submit">Play Again</button>
        <button id="login" type = "submit">Log In</button>
        </div>
        </div>
        </>
    )
}
}
export default Levels
