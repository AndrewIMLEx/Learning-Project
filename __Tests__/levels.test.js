import React from 'react';
import Levels from '../src/Homepage/Levels';
import renderer from 'react-test-renderer';

describe("Levels page",() =>{
  it('renders the levels page', () => {
    const tree = renderer.create(<Levels />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});

describe("Standard Number",() =>{
  test ('calls its callback with a random number',() =>{
    function standardNum(fn) {
      return fn(Math.floor (Math.random() *10));
    };
    let mock = jest.fn();
      standardNum(mock);
      expect(mock).toBeCalledWith(expect.any(Number));
  })
});

describe("Difficult Number",() =>{
  test('calls its callback with a random number',() =>{
    function difficultNum(fn) {
      return fn(Math.floor (Math.random() *100));
    };
    let mock = jest.fn();
      difficultNum(mock);
      expect(mock).toBeCalledWith(expect.any(Number));
  });  
});

describe("Blank Input",() =>{
  test('feedback on blank input',() =>{
    testData = [
      {guessedValue:"",Result:"Enter valid number"},
      {guessedValue: -1,Result:"Enter valid number"},
      {guessedValue: 101,Result:"Enter valid number"}
    ]
    });
    const testData = require(setname);

    testData.forEach((testItem) => {
      expect(setName(testItem.guessedValue)).toBe(testItem.Result);
    });
});

/*describe("Standard Level",() =>{
  test('Standard comparison statement',() =>{
    const standardNum = Math.floor (Math.random() *10);

    })
  })*/

  /*describe("Difficult Level",() =>{
    test('Difficult comparison statement',() =>{
      const difficultNum = Math.floor (Math.random() *10);
  
      })
    })*/
