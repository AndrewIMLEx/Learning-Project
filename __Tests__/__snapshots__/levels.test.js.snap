// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Levels page renders the levels page 1`] = `
[
  <div
    className="levelselect"
  >
    <h5
      style={
        {
          "color": "white",
          "fontSize": "17px",
          "marginLeft": "4.5em",
        }
      }
    >
      Select Difficulty Level
    </h5>
    <button
      id="standard"
      onClick={[Function]}
    >
      Standard
    </button>
    <button
      id="difficult"
      onClick={[Function]}
    >
      Difficult
    </button>
  </div>,
  <div
    className="levelCard"
  >
    <h3
      style={
        {
          "color": "black",
          "fontSize": "20px",
          "marginLeft": "5.1em",
          "paddingTop": "2em",
        }
      }
    >
      Selected Level
    </h3>
     
    <p
      style={
        {
          "color": "red",
          "fontSize": 20,
          "marginLeft": "6.3em",
        }
      }
    />
     
    <form
      id="LevelForm"
    >
      <input
        id="Levelsinput"
        max="100"
        min="0"
        onChange={[Function]}
        placeholder="Enter Number"
        style={
          {
            "float": "center",
            "fontSize": 14,
          }
        }
        type="number"
        value={[]}
      />
    </form>
     
    <p
      style={
        {
          "color": "red",
          "fontSize": 20,
          "marginLeft": "4.2em",
        }
      }
    />
       
    <button
      id="guess"
      onClick={[Function]}
      type="submit"
    >
      Guess
    </button>
     
    <div
      className="playButtons"
    >
      <button
        id="playAgain"
        onClick={[Function]}
        type="submit"
      >
        Play Again
      </button>
      <button
        id="login"
        type="submit"
      >
        Log In
      </button>
    </div>
  </div>,
]
`;
